src/plugins/discover/public/application/components/discover_grid/discover_grid.tsx

```diff
@@ -286,6 +319,18 @@ export const DiscoverGrid = ({
     [usedSelectedDocs, isFilterActive, rows, setIsFilterActive]
   );
 
+  if (isLoading) {
+    return (
+      <div className="euiDataGrid__loading">
```

Comment:
Thx a lot for implementing the loading indicator, 1 add-on here. It should only be displayed, when there are no results (This would change the behavior of the main table, if there are already results, data is refetched, there is no loading indicator in the table ... we might display a little spinner maybe in the toolbar, but this would be out of scope of this PR
```suggestion
  if (!rowCount && isLoading) {
    return (
      <div className="euiDataGrid__loading">
```